Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -34,12 +34,457 @@
 #include "simbody/internal/GeneralForceSubsystem.h"
 #include "simbody/internal/SimbodyMatterSubsystem.h"
 #include "simbody/internal/MultibodySystem.h"
+#include <iostream>
+#include <exception>
 
 #include "ForceImpl.h"
 
+#include <memory>
+
+//Threading constants used by CalcForcesTask
+namespace {
+const int NumNonParallelThreads = 1;
+const int NonParallelForcesIndex = 0;
+}
 
 namespace SimTK {
 
+/** Calculates each enabled force's contribution in the MultibodySystem.
+CalcForcesTask allows force calculations to occur in parallel with non-parallel
+forces being calculated on Thread NonParallelForcesIndex and all other parallel
+forces being calculated on seperate threads.*/
+class CalcForcesTask : public ParallelExecutor::Task {
+public:
+    //By default, CalcForcesTask will be in Mode All.
```

Comment:
I don't see how this comment is enforced. Isn't this being done by derived classes?